rts/System/Platform/CpuTopologyCommon.cpp

```diff
@@ -0,0 +1,54 @@
+#include "CpuTopology.h"
+#include <atomic>
+#include <bitset>
+#include <mutex>
+
+namespace cpu_topology {
+
+static std::mutex cacheMutex;
+static volatile bool cacheActive = false;
+
+static ProcessorMasks cachedProcessorMasks;
+static int logicalCpuCount = 0;
+static int physicalCpuCount = 0;
+
+void SetCpuCounts(ProcessorMasks& masks) {
+	const uint32_t logicalCountMask = (masks.efficiencyCoreMask & masks.performanceCoreMask);
+	const uint32_t coreCountMask = logicalCountMask & ~masks.hyperThreadHighMask;
+
+	logicalCpuCount = std::bitset<32>(logicalCountMask).count();
+	physicalCpuCount = std::bitset<32>(coreCountMask).count();
+}
+
+void InitTopologicalData() {
+	if (cacheActive)
+		return;
```

Comment:
AFAIK this sort of double-checking with a part outside the mutex is technically unsafe (see [paper](https://www.aristeia.com/Papers/DDJ_Jul_Aug_2004_revised.pdf)) but I expect it to be okay in practice so I'm not necessarily asking to do anything about it. Perhaps just
```suggestion
	if (cacheActive)
		return; // technically unsafe if outside the mutex, should be safe in practice
```

Reply:
I think std::atomic<bool> will address this.